client/src/templates/Challenges/redux/code-storage-epic.js

```diff
@@ -47,17 +47,20 @@ function getLegacyCode(legacy) {
   }, null);
 }
 
-function legacyToFile(code, files, key) {
-  if (isFilesAllPoly(files)) {
-    return { [key]: setContent(code, files[key]) };
+function legacyToFile(code, challengeFiles, fileKey) {
+  if (isFilesAllPoly(challengeFiles)) {
+    return {
+      ...setContent(
+        code,
+        challengeFiles.find(x => x.fileKey === fileKey)
+      )
+    };
   }
   return false;
 }
 
-function isFilesAllPoly(files) {
-  return Object.keys(files)
-    .map(key => files[key])
-    .every(file => isPoly(file));
+function isFilesAllPoly(challengeFiles) {
+  return challengeFiles.every(file => isPoly(file));
```

Comment:
```suggestion
  // TODO: figure out how challengeFiles might be null/not have .every as a
  // function
  return challengeFiles?.every(file => isPoly(file));
```
How about this @moT01 @ShaunSHamilton ?

Reply:
Yes. This should be the case. Backend challenges have `challengeFiles: null`

Reply:
Only thing is, were this TS, it would complain about returning `undefined` as opposed to a boolean.

Reply:
Well, I'm fine with whatever is decided - I was just pointing out my findings. If we want to just merge this as it is - let me know.

Reply:
oh geez - might have been worth it to leave it just so we didn't have to run the tests again.